compose/snippets/src/main/java/com/example/compose/snippets/pictureInPicture/PictureInPictureSnippets.kt

```diff
@@ -0,0 +1,334 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.toAndroidRectF
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalContext
+import androidx.core.app.PictureInPictureModeChangedInfo
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.toRect
+import androidx.core.util.Consumer
+import androidx.media3.common.Player
+import androidx.media3.exoplayer.ExoPlayer
+
+var shouldEnterPipMode by mutableStateOf(false)
+
+// [START android_compose_pip_broadcast_receiver_constants]
+// Constant for broadcast receiver
+const val ACTION_BROADCAST_CONTROL = "broadcast_control"
+
+// Intent extras for broadcast controls from Picture-in-Picture mode.
+const val EXTRA_CONTROL_TYPE = "control_type"
+const val EXTRA_CONTROL_PLAY = 1
+const val EXTRA_CONTROL_PAUSE = 2
+// [END android_compose_pip_broadcast_receiver_constants]
+
+@Composable
+fun PiPBuilderSetAutoEnterEnabled(
+    modifier: Modifier = Modifier
+) {
+    // [START android_compose_pip_builder_auto_enter]
+    val context = LocalContext.current
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val pipModifier = modifier.onGloballyPositioned { layoutCoordinates ->
+            val builder = PictureInPictureParams.Builder()
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                builder.setAutoEnterEnabled(true)
+            }
+            context.findActivity().setPictureInPictureParams(builder.build())
+        }
+    } else {
+        Log.i("PiP info", "API does not support PiP")
+    }
+    // [END android_compose_pip_builder_auto_enter]
+}
+
+// [START android_compose_pip_find_activity]
+internal fun Context.findActivity(): ComponentActivity {
+    var context = this
+    while (context is ContextWrapper) {
+        if (context is ComponentActivity) return context
+        context = context.baseContext
+    }
+    throw IllegalStateException("Picture in picture should be called in the context of an Activity")
+}
+// [END android_compose_pip_find_activity]
+
+@Composable
+fun EnterPiPThroughButton() {
+    // [START android_compose_pip_button_click]
+    val context = LocalContext.current
+    Button(onClick = {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            context.findActivity().enterPictureInPictureMode(
+                // the parameters have been set by previous calls
+                PictureInPictureParams.Builder().build()
+            )
+        } else {
+            Log.i("PiP info", "API does not support PiP")
+        }
+    }) {
+        Text(text = "Enter PiP mode!")
+    }
+    // [END android_compose_pip_button_click]
+}
+
+// [START android_compose_pip_is_in_pip_mode]
+@Composable
+fun isInPipMode(): Boolean {
```

Comment:
Just catching this now - I think this needs to be prefixed with remember (i.e. `rememberIsInPipMode`). See guidance here: https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-api-guidelines.md#naming-composable-functions-that-remember--the-objects-they-return

Reply:
Not resolved, this is a good catch. We have to make sure we follow these guidelines